scripts/cat/cats.py

```diff
@@ -1794,6 +2003,92 @@ def get_relatives(self, cousin_allowed=True) -> list:
             return self.inheritance.all_involved
         return self.inheritance.all_but_cousins
 
+    def get_distant_kin(self, max_depth=2):
+        """
+        Recursively retrieves distant relatives up to a specified depth,
+        excluding close kin, kin, and in-laws.
+        
+        :param max_depth: The maximum depth to explore the relative tree. Default is 2.
+        :return: A list of ids of all distant relatives up to the specified depth.
+        """
+        if not self.inheritance:
+            self.inheritance = Inheritance(self)
+        
+        def recursive_get_relatives(cat, depth):
+            if depth >= max_depth:
+                return set()
+            
+            relatives = set(cat.get_relatives())
+            new_relatives = set()
+            
+            for relative_id in relatives:
+                relative_cat = Cat.all_cats.get(relative_id)
+                if relative_cat:
+                    # don't add in-laws or cousins' cousins as they cause possible distant_kin
+                    # to balloon out of control with older, larger Clans
+                    sibling_mates = [
+                        mate_id for sibling_id in self.get_siblings()
+                        for mate_id in Cat.all_cats.get(sibling_id).mate
```

Comment:
I'm getting a crash when I try to run this PR, seems to be caused by this line
![image](https://github.com/user-attachments/assets/06572844-e5b4-4f83-880c-5ceb78edef97)